File: 20250420/leetcode781. 森林中的兔子/main.cpp
#include <cmath>
#include <unordered_map>
#include <vector>

using namespace std;
class Solution {
public:
  int numRabbits(vector<int> &answers) {
    // 相同回答的可以统计为同一种兔子
    unordered_map<int, int> hash;
    int n = answers.size();
    for (int i = 0; i < n; i++) {
      hash[answers[i]]++;
    }
    int res = 0;
    for (auto iter = hash.begin(); iter != hash.end(); iter++) {
      float first = iter->first;
      float second = iter->second;
      res = res + int(ceil(second / (first + 1))) * (first + 1);
    }
    return res;
  }
};

File: 20250420/leetcode781. 森林中的兔子/main.js
/**
 * @param {number[]} answers
 * @return {number}
 */
var numRabbits = function (answers) {
  const hash = new Map();
  for (const ans of answers) {
    hash.set(ans, (hash.get(ans) || 0) + 1);
  }
  let res = 0;
  for (const [key, val] of hash) {
    res += Math.ceil(val / (key + 1)) * (key + 1);
  }
};
